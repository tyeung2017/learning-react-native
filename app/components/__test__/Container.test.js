import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import Container, { styles } from '../Container';

it('it rendererd successfully', () => {
  const rendered = renderer.create(<Container />).toJSON();
  expect(rendered).toBeTruthy();
});

it('styles is an object', () => {
  expect(typeof styles).toBe('object');
});

it('render a child component', () => {
  const rendered = renderer.create(<Container><View />u</Container>).toJSON();
  expect(rendered).toMatchSnapshot();
});

it('styling background color', () => {
  const rendered = renderer.create(<Container backgroundColor="red" />).toJSON();
  expect(rendered).toMatchSnapshot();
});
