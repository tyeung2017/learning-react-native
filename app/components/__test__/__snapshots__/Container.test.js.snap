// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a child component 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      undefined,
    ]
  }
  testID={undefined}
>
  <View />
  u
</View>
`;

exports[`styling background color 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "red",
      },
    ]
  }
  testID={undefined}
/>
`;
